packs.go

```diff
@@ -37,44 +36,138 @@ func fileExists(filename string) bool {
 	return !info.IsDir()
 }
 
+// After a new pack download from upstream, update packs.json with installed packs
+func updatePacksInfo(langCode string, pack *Pack, packVersion *PackVersion) error {
+	packs, err := getPacksInfo()
+	if err != nil {
+		return err
+	}
+
+	var (
+		existingPackIndex int
+		existingPack      *Pack = nil
```

Comment:
What is the advantage of using pointers here. ?

Use normal variables, not pointers. Go's garbage collector will free the memory.

Reply:
It can't hold the value `nil` if it's not a pointer. Is there a better way to do this without using pointers ?

Reply:
(structName{})